src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -37,10 +37,24 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
         private readonly LspServices _lspServices;
 
         /// <summary>
-        /// The solution state that the request should operate on, if the handler requires an LSP solution, or <see langword="null"/> otherwise
+        /// The workspace this request is for, if applicable.  This will be present if <see cref="Document"/> is
+        /// present.  It will be <see langword="null"/> if <c>requiresLSPSolution</c> is false.
+        /// </summary>
+        public readonly Workspace? Workspace;
```

Comment:
Could this be obsoleted or otherwise indicated it really shouldn't be used?  I expect most instances of accessing the workspace in a request handler should not really be needed (and indeed could be dangerous)